builder/simplify.cc

```diff
@@ -755,9 +755,9 @@ class simplifier : public node_mutator {
     } else {
       interval_expr result_info = bounds_of(op, std::move(a_info.bounds), std::move(b_info.bounds));
       if (prove_constant_true(result_info.min)) {
-        set_result(true, {{1, 1}, alignment_type()});
```

Comment:
Is this necessary? I'm pretty sure we do this in many places, so it seems like this probably could be reverted?

Reply:
For reasons that aren't clear to me, Visual C++ won't compile it as-is (and yes, we do compile this with Visual C++ in some configs). Adding an `expr(bool)` ctor overload only makes things worse. I hate this kind of voodoo, but...